### PR TITLE
Refactor onboarding

### PR DESCRIPTION
**Pull request by Marik-D**

### CODE REVIEW

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx [hunk 1]
```diff
@@ -36,37 +39,47 @@ export const Onboarding = (props: OnboardingProps) => {
...
-    const wallet = await walletService.deployFutureWallet(ensName, gasParameters.gasPrice.toString(), gasParameters.gasToken, showWalletCreationModal);
+    walletCreationService.onBalancePresent(() => modalService.hideModal());
+    const wallet = await walletCreationService.initiateCreationFlow(ensName);
     modalService.hideModal();
     props.onCreate && props.onCreate(wallet);
   };
 
+  const walletState = useProperty(walletService.stateProperty);
+  const transactionHash = useProperty(walletCreationService.deploymentTransactionHash);
+  const renderModal = () => {
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx]: Extract to a separate file and make it component for example `DeploymentModal`

Comment (cehali) [reply universal-login-react/src/ui/Onboarding/Onboarding.tsx]: 👍

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx [hunk 2]
```diff
@@ -21,6 +22,8 @@ export interface OnboardingProps {
 export const Onboarding = (props: OnboardingProps) => {
   const modalService = useModalService<ReactModalType, ReactModalProps>();
   const [walletService] = useState<WalletService>(props.walletService || new WalletService(props.sdk));
+  const [walletCreationService] = useState(() => new WalletCreationService(walletService));
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx]: Should be injected (services should ALWAYS be injected!).

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx [hunk 3]
```diff
@@ -21,6 +22,8 @@ export interface OnboardingProps {
 export const Onboarding = (props: OnboardingProps) => {
   const modalService = useModalService<ReactModalType, ReactModalProps>();
   const [walletService] = useState<WalletService>(props.walletService || new WalletService(props.sdk));
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx]: Why state? Why not planning to change service during the lifecycle of the component?
Simply:
```ts
const walletService = props.walletService || new WalletService(props.sdk);
```

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx (hunk 1 above)
Comment (marekkirejczyk) [reply universal-login-react/src/ui/Onboarding/Onboarding.tsx]: 👍

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx (hunk 3 above)
Comment (Marik-D) [reply universal-login-react/src/ui/Onboarding/Onboarding.tsx]: That would create a new instance on each re-render, which is most likely not what we want

Comment (marekkirejczyk) [reply universal-login-react/src/ui/Onboarding/Onboarding.tsx]: Good point, let's make it required props then.

**Pull request merged**